shared/test/unit/mocks/mock_navigator_moz_settings.js

```diff
@@ -72,17 +77,13 @@
       },
       then: function(cb, eb) {
         // error-handler `eb` is never used, since the mock never fails.
-        var promise = new Promise(function (resolve, reject) {
-          req.addEventListener('success', function () {
-            resolve(0);
-          });
-        });
         return promise.then(cb);
       }
     };
 
     if (!_mSyncRepliesOnly) {
       timeouts.push(setTimeout(function() {
+        resolved(0);
```

Comment:
trying to understand why we have "0" as resolved value? Why don't we have just nothing?